transmogrifier/sources/xml/datacite.py

```diff
@@ -425,6 +363,100 @@ def get_locations(cls, source_record: Tag) -> list[timdex.Location] | None:
+        descriptions = source_record.metadata.find_all("description", string=True)
+        for description in descriptions:
+            if "descriptionType" not in description.attrs:
+                logger.warning(
+                    "Datacite record %s missing required Datacite attribute "
+                    "@descriptionType",
+                    cls.get_source_record_id(source_record),
+                )
+            if description.get("descriptionType") != "Abstract":
+                yield timdex.Note(
+                    value=[str(description.string)],
+                    kind=description.get("descriptionType") or None,
+                )
+
+    @classmethod
+    def get_publishers(cls, source_record: Tag) -> list[timdex.Publisher] | None:
+        publishers = []
+        if publisher := source_record.metadata.find("publisher", string=True):
+            publishers.append(timdex.Publisher(name=str(publisher.string)))
+        else:
+            logger.warning(
+                "Datacite record %s missing required Datacite field publisher",
+                cls.get_source_record_id(source_record),
+            )
+        return publishers or None
+
+    @classmethod
+    def get_related_items(cls, source_record: Tag) -> list[timdex.RelatedItem] | None:
+        return [
+            timdex.RelatedItem(
+                uri=cls.generate_related_item_identifier_url(related_identifier),
+                relationship=related_identifier.get("relationType") or "Not specified",
+            )
+            for related_identifier in source_record.metadata.find_all(
+                "relatedIdentifier", string=True
+            )
+            if related_identifier.get("relationType") != "IsIdenticalTo"
+        ] or None
+
+    @classmethod
+    def get_rights(cls, source_record: Tag) -> list[timdex.Rights] | None:
+        return [
+            timdex.Rights(
+                description=rights.string or None,
+                uri=rights.get("rightsURI") or None,
+            )
+            for rights in source_record.metadata.find_all("rights")
+            if rights.string or rights.get("rightsURI")
+        ] or None
+
+    @classmethod
+    def get_subjects(cls, source_record: Tag) -> list[timdex.Subject] | None:
+        subjects_dict: dict[str, list[str]] = {}
+
+        for subject in source_record.metadata.find_all("subject", string=True):
+            if not subject.get("subjectScheme"):
+                subjects_dict.setdefault("Subject scheme not provided", []).append(
+                    str(subject.string)
+                )
+            else:
+                subjects_dict.setdefault(subject["subjectScheme"], []).append(
+                    str(subject.string)
+                )
+
+        return [
+            timdex.Subject(value=subject_value, kind=subject_scheme)
+            for subject_scheme, subject_value in subjects_dict.items()
+        ] or None
```

Comment:
I didn't modify this code block when I migrated it to the field method but I find it a tad clunky, refactor suggestions welcome!

Reply:
Pondered on this for a while but haven't found an alternative method. I do think the use of `setdefault` works really well here! 

The only suggestion I have is to rename `subject_value` to `subject_list` or `list_of_subjects`. The term `subject_value` implies (in my mind, at least) that it is a singular value. 🤔

Reply:
Could be a good case for `defaultdict`?

```python
from collections import defaultdict

subjects_dict = defaultdict(list)
for subject in source_record.metadata.find_all("subject", string=True):
    subjects_dict[
        subject.get("subjectScheme") or "Subject scheme not provided"
    ].append(str(subject.string))

return [
    timdex.Subject(value=subject_value, kind=subject_scheme)
    for subject_scheme, subject_value in subjects_dict.items()
] or None
```

Reply:
OOOH, +1 to @ghukill 's recommendation! Just came across this [StackOverflow thread that compares the two](https://stackoverflow.com/questions/3483520/use-cases-for-the-setdefault-dict-method).

Reply:
Great solution, thank you!